src/main/java/frc/robot/subsystems/ManipulatorSubsystem.java

```diff
@@ -42,11 +42,23 @@ public void releaseHatch()
     RobotMap.hatchHolder.set(RobotMap.HATCH_HOLDER_SERVO_RELEASE);
   }
 
+  /**
+   * Controls the manipulator using an xbox controller. X-button intakes and outtakes
+   */
   public void xboxIntakeOuttake()
   {
-    if (RobotMap.assistantDriverController.getXButton())
-      spinIntake(RobotMap.MANIPULATOR_FULL_SPEED * RobotMap.MANIPULATOR_SPEED_MODIFER);
-    else
-      spinIntake(RobotMap.MANIPULATOR_STOP_SPEED);
+    //Spins the motors when the button is pressed. THIS IS ONLY RUN ONCE UNTIL THE BUTTON IS RELEASED AND PRESSED AGAIN
+    if (RobotMap.assistantDriverController.getXButtonPressed())
+    {
+      spinIntake(RobotMap.MANIPULATOR_FULL_SPEED * RobotMap.MANIPULATOR_SPEED_MODIFER); //Spin the intake
+      RobotMap.ballInIntake = !RobotMap.ballInIntake;
+    }
+    //Only stop the motors when the button is released
+    else if (RobotMap.assistantDriverController.getXButtonReleased())
+      spinIntake(RobotMap.MANIPULATOR_STOP_SPEED); //Stop the intake from spinning
+    else if (!RobotMap.ballIntakeStop.get() && RobotMap.ballInIntake)
```

Comment:
When would ballIntakeStop.get() return false?

Reply:
it depends on how the micro switch is wired. if its wired to when depressed equals false, then we dont need the !. if its the other way around we do

Reply:
i will document this method more to show the logic and how it SHOULD work tomorrow